graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/stats/ClusterStatsApi.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog.storage.elasticsearch7.stats;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import jakarta.inject.Inject;
+import org.graylog.shaded.elasticsearch7.org.elasticsearch.client.Request;
+import org.graylog.storage.elasticsearch7.PlainJsonApi;
+import org.graylog2.rest.resources.system.indexer.responses.IndexSetStats;
+
+public class ClusterStatsApi {
+    private final ObjectMapper objectMapper;
+    private final PlainJsonApi jsonApi;
+
+    @Inject
+    public ClusterStatsApi(ObjectMapper objectMapper,
+                           PlainJsonApi jsonApi) {
+        this.objectMapper = objectMapper;
+        this.jsonApi = jsonApi;
+    }
+
+    public IndexSetStats clusterStats() {
+        final JsonNode indicesCountJson = clusterStats("indices.count");
+        final JsonNode docsCountJson = clusterStats("indices.docs.count");
+        final JsonNode sizeBytesJson = clusterStats("indices.store.size_in_bytes");
```

Comment:
This executes one HTTP request for each value. That doesn't scale.

The `filter_path` takes a comma-separated list of paths. (See [docs](https://opensearch.org/docs/2.3/opensearch/common-parameters/#filtered-responses)) So we can execute a single request to get all values.

Reply:
fixed